### PR TITLE
Refactor, simply SdkTask invocation

### PR DESCRIPTION
**Pull request by tmat**
An _Arcade SDK task_ is essentially an msbuild target in a project file that can be invoked from YAML definition. A task might need to have its dependencies restored, if executed on a build agent that did not run the main repo build. 

This change simplifies some of the invocation logic. 

- Removes `SdkTask.proj` and moves the (simplified) logic to `sdk-task.ps1`.

- Removes `eng/common/PublishBuildAssets.cmd`: the file supplied some parameters to `sdk-task.ps1` while other parameters were still required to be passed in the YAML definition. It seems simpler to pass all parameters in the definition.

- Add `Directory.Build` files to SdkTasks: all tasks have some common files to import.

- Only allow a single task to be executed by `sdk-task.ps1`. The task itself can execute multiple targets as needed.

### CODE REVIEW

**Comment by tmat** [comment]
@chcosta PTAL

File: eng/common/sdk-task.ps1 [hunk 1]
```diff
@@ -16,33 +13,29 @@ Param(
...
+    Write-Host "  -task <value>           Name of Arcade task (name of a project in SdkTasks directory of the Arcade SDK package)"
+    Write-Host "  -restore                Restore dependencies"
+    Write-Host "  -verbosity <value>      Msbuild verbosity: q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic]"
     Write-Host "  -help                   Print help and exit"
     Write-Host ""
 
     Write-Host "Advanced settings:"
-    Write-Host "  -restore                Restore dependencies (short: -r)"
-    Write-Host "  -projects <value>       Semi-colon delimited list of sln/proj's from the Arcade sdk to build. Globbing is supported (*.sln)"
     Write-Host "  -ci                     Set when running on CI server"
```
Comment (chcosta) [inline eng/common/sdk-task.ps1]: Remove "-ci"

File: eng/common/sdk-task.ps1 [hunk 2]
```diff
@@ -16,33 +13,29 @@ Param(
...
     Write-Host "  -msbuildEngine <value>  Msbuild engine to use to run build ('dotnet', 'vs', or unspecified)."
     Write-Host ""
     Write-Host "Command line arguments not listed above are passed thru to msbuild."
-    Write-Host "The above arguments can be shortened as much as to be unambiguous (e.g. -co for configuration, -t for test, etc.)."
 }
 
-function Build {
-  $toolsetBuildProj = InitializeToolset
+function Build([string]$target) {
+  $bl = if ($binaryLog) { "/bl:" + (Join-Path $LogDir "$task.$target.binlog") } else { "" }
```
Comment (chcosta) [inline eng/common/sdk-task.ps1]: `binaryLog` switch was removed, just define it

File: eng/common/sdk-task.ps1 [hunk 3]
```diff
@@ -52,17 +45,27 @@ try {
...
 
-  if ($ci) {
-    $binaryLog = $true
+  $taskProject = GetSdkTaskProject $task
+  if (!(Test-Path $taskProject)) {
+    Write-Host "Unknown task: $task" -ForegroundColor Red
+    ExitWithExitCode 1
+  }
+
+  $ci = $true
```
Comment (chcosta) [inline eng/common/sdk-task.ps1]: `$ci` is not used anywhere

**Comment by chcosta** [comment]
I like this direction better

File: eng/common/sdk-task.ps1 (hunk 1 above)
Comment (tmat) [reply eng/common/sdk-task.ps1]: Done.

File: eng/common/sdk-task.ps1 (hunk 3 above)
Comment (tmat) [reply eng/common/sdk-task.ps1]: It is still expected to be set by the code in `tools.ps1`.

Comment (tmat) [reply eng/common/sdk-task.ps1]: I guess it should actually be set before tools.ps1 is imported - not a big difference, but some defaults are initialized based on the value.

Comment (tmat) [reply eng/common/sdk-task.ps1]: I'll move it.

**Comment by nguerrera** [comment]
> Removes `eng/common/PublishBuildAssets.cmd`

This broke our builds. Were we not supposed to be referencing it?

**Comment by nguerrera** [comment]
https://github.com/dotnet/sdk/blob/6c3324d3624a10c9d1dfcfba188c974063e86984/eng/common/templates/phases/publish-build-assets.yml#L31

**Comment by nguerrera** [comment]
How should we fix this. @chcosta @tmat

**Comment by nguerrera** [comment]
Actually that's coming from arcade (it's in common): https://github.com/dotnet/arcade/blob/d07ee4384ebed552beda1686469d16e25545c252/eng/common/templates/phases/publish-build-assets.yml#L31

Seems like a bug, right?

**Comment by nguerrera** [comment]
cc @livarcocc

**Comment by chcosta** [comment]
We should probably add that file back.  I presume aspnet repos will be broken by deleting that as well.

**Comment by tmat** [comment]
I missed that one :(

Instead of adding the file back I'm gonna update the YAML like I did the other one:
https://github.com/dotnet/arcade/pull/1939/files#diff-5150f6621beb9325102a281cb90b5202R49

**Comment by tmat** [comment]
Here: https://github.com/dotnet/arcade/pull/1992

**Pull request merged**